dev/DynamicDependency/M.AM.DD.PackageDependency.cpp

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+
+#include "M.AM.DD.PackageDependency.h"
+#include "Microsoft.ApplicationModel.DynamicDependency.PackageDependency.g.cpp"
+
+#include "M.AM.DD.PackageDependencyContext.h"
+
+#include <MsixDynamicDependency.h>
+#include <wil/resource.h>
+#include <wil_msixdynamicdependency.h>
+
+#include <M.AM.Converters.h>
+
+namespace winrt::Microsoft::ApplicationModel::DynamicDependency::implementation
+{
+    PackageDependency::PackageDependency(PCWSTR id)
+    {
+        m_id = winrt::to_hstring(id);
+    }
+    PackageDependency::PackageDependency(hstring const& id) :
+        m_id(id)
+    {
+    }
+    Microsoft::ApplicationModel::DynamicDependency::PackageDependency PackageDependency::GetFromId(hstring const& id)
+    {
+        return winrt::make<implementation::PackageDependency>(id);
+    }
+    Microsoft::ApplicationModel::DynamicDependency::PackageDependency PackageDependency::Create(hstring const& packageFamilyName, Windows::ApplicationModel::PackageVersion const& minVersion)
+    {
+        const auto mddMinVersion = ::Microsoft::ApplicationModel::DynamicDependency::ToVersion(minVersion);
```

Comment:
These smell a lot like hungarian; can you just use `minVersion` and `architectures` and `lifetimeKind`, etc?

Reply:
Ambiguity

PackageDependency PackageDependency::Create(hstring const& packageFamilyName, Windows::ApplicationModel::PackageVersion const& ***minVersion***)

We need 2 variables in the same scope - WinRT + Flat-C datatype
* Windows::ApplicationModel::PackageVersion vs PACKAGE_VERSION
etc

I used the mdd prefix to indicate they're the Flat-C types

I could have skipped the prefix in some cases but then we'd have a mix of no-prefix and prefix so this was clearer

I could have gone with a suffix, e.g. minVersion vs minVersionAsPackageVersion but that'd be confusing since that's the literal WinRT type name, and minVersionAsPACKAGE_VERSION is ugly as sin

Reply:
Namespace-wise, try this:

```c++
namespace wam = winrt::Windows::ApplicationModel;
namespace mdd = winrt::Microsoft::ApplicationModel::DynamicDependency;

wam::PackageVersion foo { ... };
mdd::PackageVersion bar { ... };
```

The `mdd-` variable prefix just seems off.